### PR TITLE
openstack: delete loadbalancer resources

### PR DESCRIPTION
**Pull request by MaysaMacedo**
This PR adds support for Load Balancer deletion on cluster destroy.
It detects when tag is supported in Octavia and if it is, uses it to filter and so delete Octavia resources, otherwise uses the descriptions field to filter the resources.

### CODE REVIEW

File: pkg/destroy/openstack/openstack_deprovision.go [hunk 1]
```diff
@@ -541,6 +544,73 @@ func deleteTrunks(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fi
...
+
+	allAPIVersions, err := apiversions.ExtractAPIVersions(newallPages)
+	if err != nil {
+		logger.Debug("Unable to extract api versions: %v", err)
+	}
+
+	var octaviaTagSupport bool
+	octaviaTagSupport = false
+	for _, apiVersion := range allAPIVersions {
+		if apiVersion.ID >= "v2.5" {
```
Comment (celebdor) [inline pkg/destroy/openstack/openstack_deprovision.go]: maybe this v2.5 should come from a constant

File: pkg/destroy/openstack/openstack_deprovision.go [hunk 2]
```diff
@@ -541,6 +544,73 @@ func deleteTrunks(opts *clientconfig.ClientOpts, filter Filter, logger logrus.Fi
...
+			octaviaTagSupport = true
+		}
+	}
+
+	tags := filterTags(filter)
+	var listOpts loadbalancers.ListOpts
+	if octaviaTagSupport {
```
Comment (luis5tb) [inline pkg/destroy/openstack/openstack_deprovision.go]: what would happen if the OpenStack cloud was upgraded after the installation of OCP? Could it be that octavia was not having support for tags, and then after the upgrade it has, but the resources are not tagged but using description? Perhaps better to run the else part regardless of the octavia version?

Comment (MaysaMacedo) [reply pkg/destroy/openstack/openstack_deprovision.go]: Good point. I've updated the PR with you suggestion. Thanks!

File: pkg/destroy/openstack/openstack_deprovision.go (hunk 1 above)
Comment (MaysaMacedo) [reply pkg/destroy/openstack/openstack_deprovision.go]: Thanks, Toni.

**Comment by sdodson** [comment]
CI failures appear to be related to the code changes in this PR.

**Comment by tomassedovic** [comment]
/hold

Thanks @MaysaMacedo! This looks good to me, but IIRC the installer is now in a feature freeze, so we'll need to wait before that lifts.

@wking is that correct or can we merge this? It only affects OpenStack, but it's adding to the `vendor` dir too.

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/installer/pull/1496/commits/bf64e3de1aa65d39cb339bca286aa8764e9cf0fe
/approve

**Comment by trown** [comment]
/lgtm

CI is being extra flaky on this patch, but it made it to the installer exiting zero on one of the runs.

**Pull request merged**